test/src/meshgenerators/TestDataDrivenGenerator.C

```diff
@@ -0,0 +1,73 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TestDataDrivenGenerator.h"
+
+#include "libmesh/mesh_generation.h"
+#include "libmesh/unstructured_mesh.h"
+
+registerMooseObject("MooseTestApp", TestDataDrivenGenerator);
+
+InputParameters
+TestDataDrivenGenerator::validParams()
+{
+  auto params = MeshGenerator::validParams();
+
+  params.addParam<unsigned int>("nx", "The metadata value of nx to set, if any");
+  params.addParam<unsigned int>("ny", "The metadata value of ny to set, if any");
+
+  params.addParam<MeshGeneratorName>("nx_generator",
+                                     "The generator to get nx from (if generating a mesh)");
+  params.addParam<MeshGeneratorName>("ny_generator",
+                                     "The generator to get ny from (if generating a mesh)");
+
+  MeshGenerator::setHasGenerateData(params);
+
+  return params;
+}
+
+TestDataDrivenGenerator::TestDataDrivenGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters)
+{
+  if (isParamValid("nx"))
+    declareMeshProperty<unsigned int>("nx");
+  if (isParamValid("ny"))
+    declareMeshProperty<unsigned int>("ny");
+  if (isParamValid("nx_generator"))
+    (void)getMesh("nx_generator");
+  if (isParamValid("ny_generator"))
+    (void)getMesh("ny_generator");
```

Comment:
is this checking whether the generators specified by the parameters are valid?

Reply:
```
[Mesh]
  [nx]
    type = TestDataDrivenGenerator
    nx = 4
  []
  [ny]
    type = TestDataDrivenGenerator
    ny = 2
  []
  [mesh]
    type = TestDataDrivenGenerator
    nx_generator = nx
    ny_generator = ny
  []
  [transform]
    type = TransformGenerator
    input = mesh
    transform = translate
    vector_value = '10 5 0'
  []
  data_driven_generator = mesh
[]
```

`nx` adds a piece of data called `nx` and has no mesh. `ny` adds a piece of data called `ny` and has no mesh. `mesh` then adds a dependency on `nx` and `ny` (this is the piece you commented on; `(void)` is there because of `[[nodiscard]]`).  `mesh` then uses the metadata `nx/nx` and `ny/ny` to build a simple grid

Reply:
got ya. I thought that might be the case, but didn't see where that dependency was being added during the `getMesh` call. It was late though and my brain was definitely not running at full capacity

Reply:
it definitely isn't the most intuitive test to follow, for what it's worth

Reply:
I see now that the dependency is inserted in `getMeshByName` called from `getMesh` 👍